packages/idos-sdk-js/README.md

```diff
@@ -227,3 +260,62 @@ await idos.data.update("attributes", { id, value: "1000" });
 
 await idos.data.delete("attributes", id);
 ```
+
+
+### Access Grants
+
+An Access Grant is a record in the blockchain that consists of the following values:
+
+* `owner`: the grant owner (in ETH chain this is the owners wallet address, for Near this is the owners full access public key).
+* `grantee`: the grant grantee (in ETH chain this is the grantee wallet address, for Near this is the grantee full access public key).
+* `dataId`: the `id` of the duplicated record (i.e credential) that is going to be shared.
+* `lockedUntil`: the earliest UNIX timestamp when the contract will allow the Access Grant to be revoked. "0" means it's revocable at any time.
+
+Acquiring an access grant assures a dApp that they'll have a copy of the user's credential until `lockedUntil`. This is especially relevant to be able to fulfill compliance obligations.
+
+### Access Grant creation / revocation / list
+
+For EVM wallets, the access grant is created by invoking the `insertGrant` contract action, where the grant owner is the user who signed the transaction.
+For EVM wallets, the access grant is revoked by invoking the `deleteGrant` contract action, where the grant owner is the user who signed the transaction.
+You can en example of a contract [here](https://sepolia.etherscan.io/address/0x1673b9fd14c30332990314d99826f22a628a2601#code)
+
+```js
+// Share a credential by creating an access grant
+await idos.grants.create('credential', credential.id, grantee, timelock, receiverPublicKey)
+    
+// Revoke an access grant
+await idos.grants.revoke('credentials', recordId, grantee, dataId, timelock)
+
+// List all grants that match a criteria
+await idos.grants.list({
+    owner,
+    dataId,
+    grantee
+})
+
+```
+When creating an access grant some things happen under the hood:
+* A duplicate of the underlying data is created, encrypted to the provided `receiverPublicKey`
+* An access grant is created on the connected chain
+
+The access grant can be revoked only once its timelock has expired.
+
+> [!TIP]
+> See a working example [idos-example-dapp](../../apps/idos-example-dapp)
+
+
+### Delegated Access Grants
+
+A delegated Access Grant (dAG) is a way of creating / revoking an Access Grant by somebody else other than the user. This is acomplished by getting the user's signature a specific message, generated with the contract's `insert_grant_by_signature_message` method, that can then be used to call the contract's `insert_grant_by_signature` method.
+
+This is especially relevant for dApps who want to subsidise the cost of transaction necessary to create an AG.
+
+The interface for creating a dAG is the same as the one for creating an Access Grant:
```

Comment:
This is not true. We happen to use dAGs under the hood for NEAR over technical limitations, but this is now how you'd make a dAG for an EVM user.